### PR TITLE
update 0.4.0 with cudf 0.18.1

### PR DESCRIPTION
**Pull request by NvTimLiu**
Merge source code from branch branch-0.4 to main

### CODE REVIEW

**Comment by NvTimLiu** [comment]
build

**Review by pxLi — APPROVED** [review]
LGTM

**NOTE: this should be merged as a merge commit**

**Comment by NvTimLiu** [comment]
> LGTM
> 
> **NOTE: this should be merged as a merge commit**

Yes, as a merge commit instead of squashing

File: docs/demo/Databricks/generate-init-script.ipynb [hunk 1]
```diff
@@ -1 +1 @@
...
+{"cells":[{"cell_type":"code","source":["dbutils.fs.mkdirs(\"dbfs:/databricks/init_scripts/\")\n \ndbutils.fs.put(\"/databricks/init_scripts/init.sh\",\"\"\"\n#!/bin/bash\nsudo wget -O /databricks/jars/rapids-4-spark_2.12-0.4.0.jar https://oss.sonatype.org/content/repositories/staging/com/nvidia/rapids-4-spark_2.12/0.4.0/rapids-4-spark_2.12-0.4.0.jar\nsudo wget -O /databricks/jars/cudf-0.18.1-cuda10-1.jar https://oss.sonatype.org/content/repositories/staging/ai/rapids/cudf/0.18.1/cudf-0.18.1-cuda10-1.jar\"\"\", True)"],"metadata":{},"outputs":[],"execution_count":1},{"cell_type":"code","source":["%sh\ncd ../../dbfs/databricks/init_scripts\npwd\nls -ltr\ncat init.sh"],"metadata":{},"outputs":[],"execution_count":2},{"cell_type":"code","source":[""],"metadata":{},"outputs":[],"execution_count":3}],"metadata":{"name":"generate-init-script","notebookId":2645746662301564},"nbformat":4,"nbformat_minor":0}
```
Comment (GaryShen2008) [inline docs/demo/Databricks/generate-init-script.ipynb:1]: Seems here is using a stage url. https://oss.sonatype.org/content/repositories/staging/ai/rapids/cudf/0.18.1/cudf-0.18.1-cuda10-1.jar. Let's fix it later.

**Review by GaryShen2008 — APPROVED** [review]

**Pull request merged**